java/src/main/java/io/cucumber/java/DefaultDataTableEntryTransformer.java

```diff
@@ -25,5 +25,5 @@
 @Target(ElementType.METHOD)
 @API(status = API.Status.STABLE)
 public @interface DefaultDataTableEntryTransformer {
-
+    boolean titleCaseHeaders() default false;
```

Comment:
As this will go into the v5 release and as `@DefaultDataTableEntryTransformer` will be introduced in the same realse we can set this to true and provide this as a new feature by default. That way we'll require minimal amounts of configuration to get a working system.

I'm not sure about the name. We're not creating title case headers. We're turning headers into property names. Maybe `convertHeadersToProperties`?  I don't have a good suggestion right now though.